lib/new_relic/agent/method_tracer_helpers.rb

```diff
@@ -68,10 +68,10 @@ def clm_enabled?
       end
 
       # The string representation of a singleton class looks like
-      # '#<Class:MyModule::MyClass>'. Return the 'MyModule::MyClass' part of
-      # that string
+      # '#<Class:MyModule::MyClass>', or '#<Class:MyModule::MyClass(id: integer, attrbiute: string)>'
+      # Return the 'MyModule::MyClass' part of that string
       def klass_name(object)
-        name = Regexp.last_match(1) if object.to_s =~ /^#<Class:(.*)>$/
+        name = Regexp.last_match(1) if object.to_s =~ /^#<Class:([\w:]*).*>$/
```

Comment:
With `[\w:]`, you can end up with colons in the match text. Any need to strip those away?

Reply:
I added the colon because we needed it in order to handle a class name like `Something::Class`, otherwise it would only match `Something` instead of the full thing, which would not be the correct name to use. So we would want the colons in the name to make sure our `Object.const_get` works correctly.